qemu/qmp.go

```diff
@@ -883,11 +883,20 @@ func (q *QMP) ExecHotplugMemory(ctx context.Context, qomtype, id, mempath string
 }
 
 // ExecutePCIVSockAdd adds a vhost-vsock-pci bus
-func (q *QMP) ExecutePCIVSockAdd(ctx context.Context, id, guestCID string) error {
+func (q *QMP) ExecutePCIVSockAdd(ctx context.Context, id, guestCID, vhostfd string, disableModern bool) error {
 	args := map[string]interface{}{
 		"driver":    VHostVSockPCI,
 		"id":        id,
 		"guest-cid": guestCID,
 	}
+
+	if vhostfd != "" {
+		args["vhostfd"] = vhostfd
+	}
+
+	if disableModern {
+		args["disable-modern"] = disableModern
```

Comment:
I am fine with this being part of the same commit, but please change the commit message by replacing `of "modern" devices, it's used to make azure VMs happy.` with `of "modern" devices, by using virtio 0.9 instead of virtio 1.0. Particularly, this is useful when running the VM in a nested environment.`

Reply:
done